File: app/src/main/java/ru/job4j/weatherkt/model/City.java
package ru.job4j.weatherkt.model;

import com.google.gson.annotations.SerializedName;

import java.util.Objects;

/**
 * @author Dmitry Kolganov (mailto:dmk78@inbox.ru)
 * @version $Id$
 * @since 01.12.2019
 */

public class City {


    private Integer id;

    private String name;

    private Coord coord;

    private String country;

    private Integer population;
    @SerializedName("timezone")

    private Integer timeZone;

    private Integer sunrise;

    private Integer sunset;

    public Integer getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Coord getCoord() {
        return coord;
    }

    public String getCountry() {
        return country;
    }

    public Integer getPopulation() {
        return population;
    }

    public Integer getTimeZone() {
        return timeZone;
    }

    public Integer getSunrise() {
        return sunrise;
    }

    public Integer getSunset() {
        return sunset;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        City city = (City) o;
        return Objects.equals(id, city.id) &&
                Objects.equals(name, city.name) &&
                Objects.equals(coord, city.coord) &&
                Objects.equals(country, city.country) &&
                Objects.equals(population, city.population) &&
                Objects.equals(timeZone, city.timeZone) &&
                Objects.equals(sunrise, city.sunrise) &&
                Objects.equals(sunset, city.sunset);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, coord, country, population, timeZone, sunrise, sunset);
    }
}

File: app/src/main/java/ru/job4j/weatherkt/model/City.kt
package ru.job4j.weatherkt.model

import com.google.gson.annotations.SerializedName

/**
 * @author Dmitry Kolganov (mailto:dmk78@inbox.ru)
 * @version $Id$
 * @since 01.12.2019
 */
data class City (
    val id: Int? = null,
    val name: String? = null,
    val coord: Coord? = null,
    val country: String? = null,
    val population: Int? = null,
    @SerializedName("timezone")
    val timeZone: Int? = null,
    val sunrise: Int? = null,
    val sunset: Int? = null
)
